src/app/shell/result/result.component.ts

```diff
@@ -1,22 +1,39 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, OnDestroy } from '@angular/core';
 import { UpdateCurrentView } from '../../shared/result.actions';
 import { Store } from '@ngxs/store';
 import { ChangePage } from 'src/app/shared/store/app.actions';
+import { Nav } from 'src/app/shared/models/navigation.model';
+import { AddNavPath, DeleteNavPath } from 'src/app/shared/store/navigation.actions';
 
 @Component({
   selector: 'app-result',
   templateUrl: './result.component.html',
   styleUrls: ['./result.component.scss']
 })
-export class ResultComponent implements OnInit {
+export class ResultComponent implements OnInit, OnDestroy {
   public currentView: string;
   isFiltersVisible: boolean = true;
 
   constructor(private store: Store) { }
 
+  /**
+    * This method create new Navigation button
+    */
+  creatNavPath(name:string, isActive: boolean, disable: boolean): Nav[] {
```

Comment:
I can see that createNavPath is duplicated in each component, I believe it will be better to move it to the service